### ISSUE TITLE
PocketTTS cloned voices are clipped/clicky: conditioning strides and bos_before_voice

### ISSUE THREAD

**Issue by kkilchrist** [FluidInference/FluidAudio#612]
## Summary

PocketTTS voice cloning in FluidAudio currently produces clipped/clicky or phone-like output for cloned voices. I reproduced this through `fluidaudiocli` and compared the Swift implementation against upstream `kyutai-labs/pocket-tts`.

I think there are two related issues in the Swift/CoreML clone path:

1. `PocketTtsVoiceCloner.extractConditioning` assumes the CoreML `conditioning` `MLMultiArray` is contiguous for `.float32` and bulk-copies from `dataPointer`. The `conditioning` array can be strided/non-contiguous, so this reads the wrong embedding order.
2. Upstream PocketTTS prepends `bos_before_voice` before cloned voice prompt frames when `flow_lm.insert_bos_before_voice` is true. FluidAudio ships/has `constants_bin/bos_before_voice.bin`, but cloned voices returned by `PocketTtsVoiceCloner` appear to contain only raw Mimi conditioning frames.

## Environment

- FluidAudio: `0.14.5`
- Backend: PocketTTS via `fluidaudiocli`
- Platform: Apple Silicon macOS
- Language pack: `english`

## Repro

Generate a baseline, then clone from that baseline or another clear WAV:

```sh
swift run fluidaudiocli tts \
  'Hello. This is a local voice cloning test using Pocket TTS.' \
  --backend pocket \
  --language english \
  --voice alba \
  --output /tmp/pocket_builtin.wav

swift run fluidaudiocli tts \
  'Hello. This is a local voice cloning test using Pocket TTS.' \
  --backend pocket \
  --language english \
  --clone-voice /tmp/pocket_builtin.wav \
  --save-voice /tmp/pocket_clone.bin \
  --output /tmp/pocket_clone.wav
```

With a longer real voice sample, I saw logs like:

```text
Encoded 547200 samples (~22.7s), 285 frames, using 250
Generated 28 frames
Audio duration: 2.24s
```

The generated cloned-voice audio was clipped/clicky and did not sound like the prompt. A similar failure reproduced on-device from an iOS app using FluidAudio.

### 1. `MLMultiArray` stride handling

Current released code effectively does this for `.float32` conditioning:

```swift
let srcPtr = conditioning.dataPointer.bindMemory(to: Float.self, capacity: count)
return Array(UnsafeBufferPointer(start: srcPtr, count: count))
```

That assumes the desired `[1, frames, embDim]` slice is contiguous in row-major order. In local testing, replacing it with stride-aware extraction changed the cloned voice behavior substantially.

Patch shape:

```swift
let strides = conditioning.strides.map { $0.intValue }
let frameStride = strides[1]
let dimStride = strides[2]
let pointer = conditioning.dataPointer.bindMemory(to: Float.self, capacity: conditioning.count)

for frame in 0..<frames {
    let frameBase = frame * frameStride
    for dim in 0..<embDim {
        values.append(pointer[frameBase + dim * dimStride])
    }
}
```

This should fall back to indexed `conditioning[[0, frame, dim]].floatValue` if shape/stride validation fails.

### 2. Missing `bos_before_voice` for cloned voices

In upstream `kyutai-labs/pocket-tts`, `pocket_tts/models/tts_model.py` does:

```python
prompt = self._encode_audio(audio_conditioning.unsqueeze(0).to(self.device))

if self.flow_lm.insert_bos_before_voice:
    prompt = torch.cat([self.flow_lm.bos_before_voice, prompt], dim=1)
```

And `pocket_tts/config/english.yaml` has:

```yaml
flow_lm:
  insert_bos_before_voice: true
```

FluidAudio's downloaded English pack has:

```text
constants_bin/bos_before_voice.bin  # 1024 Float32 values
```

but `PocketTtsConstantsLoader` loads `bos_emb.bin` and `text_embed_table.bin`, not `bos_before_voice.bin`, and `PocketTtsVoiceCloner` returns `PocketTtsVoiceData(audioPrompt: voiceData, promptLength: usableFrames)` without prepending BOS.

In local testing:

- A stride-safe cloned prompt without BOS generated a clipped short utterance for an app-like sample window: ~17 frames / 1.36s.
- The same prompt with `bos_before_voice.bin` prepended, total prompt length 126 tokens, generated a normal-length utterance: ~48 frames / 3.84s for the same text.
- Prepending BOS while leaving total prompt length near 250 behaved poorly in the CoreML path, so matching shipped voice snapshot length (`1 + PocketTtsConstants.voicePromptLength`, i.e. 126 tokens) may be the safer cap unless there is a better expected prompt length.

## Expected

`PocketTtsVoiceCloner.cloneVoice` should produce cloned `PocketTtsVoiceData` that can be passed to `PocketTtsManager.synthesize(text:voiceData:)` and generate stable, normal-duration speech comparable to the upstream Python behavior.

## Suggested fix

- Make `extractConditioning` stride-aware for `.float32` `MLMultiArray`s.
- Load `constants_bin/bos_before_voice.bin` into PocketTTS constants, or otherwise make it available to the clone path.
- Prepend `bos_before_voice` to cloned Mimi conditioning when the language pack expects it.
- Consider capping cloned prompt frames to `PocketTtsConstants.voicePromptLength` audio frames plus BOS, or validating what prompt length the exported CoreML FlowLM path supports reliably.

I have a local app-side workaround with these changes and can share a patch if useful.

**Comment by Alex-Wengg** [comment]
> I have a local app-side workaround with these changes and can share a patch if useful.

yes please
